code/script.js

```diff
@@ -249,95 +251,147 @@ const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
 }
 
+
+
 // This function to start (and restart) the game
 const start = () => {
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
   // What else should happen when we start the game?
+  //1. Set the secret person 
+  setSecret()
+  // 2. Cards generated
+  generateBoard()
+
+   //Hide the winOrLose section after pressing the PLAY AGAIN button 
+   winOrLose.style.display = "none"
+   // Showing back the board 
+   board.style.display = "flex"
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
+
   // This variable stores what option group (category) the question belongs to.
-  // We also need a variable that stores the actual value of the question we've selected.
+  const category = questions.options[questions.selectedIndex].parentNode.label
+  
+  // Variable that stores the actual value of the question we've selected.
+  const value = questions.value
+  
 
   if (category === 'hair color') {
     currentQuestion = {
       attribute: 'hairColor',
-      // value: ,
-      // 👆 add the value from the input here
+      value: value,
       category: category,
     }
   } else if (category === 'eye color') {
-    // Set this up your self
+    currentQuestion = {
+      attribute: 'eyeColor',
+      value: value,
+      category: category,
+    }
   } else if (category === 'accessories') {
     currentQuestion = {
-      //attribute: ,
+      attribute: value,
       // 👆 this is the property of the booleans such as smoke, glasses and hat. add the value from the input here
       value: true, // we're asking if this person wears a hat for exaple, so always true in the question.
       category: category,
     }
   } else if (category === 'other') {
-    // Set this up your self (should be same structure as above)
+    currentQuestion = {
+      attribute: value,
+      value: true,
+      category: category,
+   }
   }
 }
 
 // This function should be invoked when you click on 'Find Out'.
```

Comment:
It's a little bit messy to read with all the teacher's instructions left in the code. It would be great to have your own comments to the code instead to understand the thinking process and for the sake of readability.